# Assignment 1 
- Implement the sorting algorithm you came up with in pseudocode with Python.
- Test the sorting algorithm with a list of 10, 100, 1000 random numbers and compare the result using the %time to time your code and submit your results in code comments.

In [188]:
import random

## Pseudocode

Read in the list and count the number of its elements. Build a loop that runs as many times. 

Build a second loop, starting at the index of the first element plus 1, runing for the full length of the list.

Check if the selected element is smaller than the one from the first loop. If it is: Exchange the two elements.

## The algorithm

In [189]:
def ownsort(int_list):
    for i in range(len(int_list) - 1):
        k = i
        for j in range(i + 1, len(int_list)):
            if int_list[j] < int_list[k]:
                k = j
        int_list[i], int_list[k] = int_list[k], int_list[i]
    return int_list

In [190]:
int_list = [3,1,6,55,13,22,6,9,32]
ownsort(int_list)

[1, 3, 6, 6, 9, 13, 22, 32, 55]

## Test the sorting Algorithm

In [191]:
# Generate random numbers

int_list_10 = []
for i in range(0,10):
    int_list_10.append(random.randint(1,1000))

int_list_100 = []
for i in range(0,100):
    int_list_100.append(random.randint(1,1000))

int_list_1000 = []
for i in range(0,1000):
    int_list_1000.append(random.randint(1,1000))

In [192]:
print(ownsort(int_list_10))
%time

[246, 273, 286, 389, 426, 637, 801, 954, 974, 996]
CPU times: user 0 ns, sys: 0 ns, total: 0 ns
Wall time: 5.72 µs


Result of %time:

CPU times: user 0 ns, sys: 0 ns, total: 0 ns
Wall time: 5.72 µs

In [194]:
print(ownsort(int_list_100))
%time

[11, 13, 31, 47, 73, 74, 76, 95, 105, 110, 135, 151, 154, 161, 165, 165, 166, 177, 177, 184, 189, 190, 213, 220, 220, 249, 266, 267, 290, 291, 298, 307, 318, 323, 337, 343, 346, 348, 348, 348, 360, 367, 371, 414, 452, 472, 476, 477, 483, 505, 510, 526, 546, 557, 575, 575, 587, 592, 599, 614, 619, 641, 662, 665, 677, 677, 691, 696, 699, 699, 710, 720, 740, 745, 754, 759, 767, 770, 782, 791, 794, 796, 815, 826, 836, 836, 839, 884, 885, 912, 931, 932, 942, 944, 951, 952, 962, 979, 983, 987]
CPU times: user 0 ns, sys: 0 ns, total: 0 ns
Wall time: 5.48 µs


Result of %time:

CPU times: user 0 ns, sys: 0 ns, total: 0 ns
Wall time: 5.48 µs  

In [195]:
print(ownsort(int_list_1000))
%time

[2, 4, 5, 7, 8, 8, 9, 9, 10, 12, 15, 15, 15, 16, 16, 17, 17, 18, 18, 18, 19, 19, 19, 20, 21, 21, 22, 23, 23, 24, 25, 27, 27, 30, 31, 31, 33, 33, 34, 37, 39, 42, 42, 43, 43, 43, 43, 45, 46, 46, 47, 47, 48, 49, 50, 52, 52, 53, 53, 54, 54, 56, 56, 56, 59, 61, 61, 63, 63, 65, 66, 66, 67, 68, 68, 68, 70, 70, 72, 72, 73, 73, 74, 76, 76, 78, 79, 79, 79, 81, 82, 82, 82, 82, 83, 83, 84, 84, 85, 85, 86, 86, 87, 88, 88, 89, 92, 92, 93, 93, 93, 94, 95, 96, 97, 98, 99, 101, 105, 106, 108, 110, 111, 112, 116, 117, 118, 119, 119, 120, 121, 124, 127, 127, 128, 128, 129, 131, 133, 133, 133, 135, 136, 136, 136, 138, 138, 139, 144, 145, 145, 145, 148, 148, 148, 148, 148, 148, 149, 151, 152, 153, 153, 157, 158, 160, 161, 162, 163, 164, 167, 168, 168, 169, 169, 170, 171, 172, 172, 174, 174, 175, 175, 176, 176, 177, 177, 178, 179, 182, 184, 185, 185, 185, 186, 188, 189, 189, 190, 190, 191, 192, 193, 193, 195, 197, 197, 199, 201, 201, 202, 203, 203, 205, 206, 206, 207, 208, 210, 211, 213, 213, 214, 215, 217,

Result of %time:

CPU times: user 0 ns, sys: 0 ns, total: 0 ns
Wall time: 4.05 µs